specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json

```diff
@@ -0,0 +1,1045 @@
+				"description": "This API gets the current quota limit and usages for the specific resource for resource provider for the specified location. This response can be used to submit quotaRequests.",
+				"operationId": "Quota_listStatus",
+				"x-ms-examples": {
+					"Quotas_Request_ForCompute": {
+						"$ref": "./examples/getComputeOneSkuUsages.json"
+					}
+				},
+				"parameters": [
+					{
+						"$ref": "#/parameters/SubscriptionIdInParameters"
+					},
+					{
+						"$ref": "#/parameters/ProviderIdInParameters"
+					},
+					{
+						"$ref": "#/parameters/LocationIdInParameters"
+					},
+					{
+						"$ref": "#/parameters/ApiVersionParameter"
+					},
+					{
+						"$ref": "#/parameters/ResourceNameInParameters"
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "OK. Quota information will be returned as response with pagination.",
+						"schema": {
+							"$ref": "#/definitions/CurrentQuotaLimit"
+						},
+						"headers": {
+							"ETag": {
+								"description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+								"type": "string"
+							}
+						}
+					},
+					"default": {
+						"description": "Error response describing why the operation failed.",
+						"schema": {
+							"$ref": "#/definitions/ExceptionResponse"
+						}
+					}
+				}
+			},
+			"put": {
+				"tags": [
+					"QuotaRequests"
+				],
+				"summary": "Submits a Quota Request for a resource provider at the specified location for the specific resource in the parameter.",
+				"description": "Submits Quota change request for a resource provider for the specified location for the specific resource in the parameter. To use, first make a Get request to get quota information. This information consists of a list of resources and information regarding those resources. For all the resources in that list which require an update to their quotas, update their limit fields in the response from the Get request to their new values. Then, submit this updated JSON object to this quota request API. This will update the quotas to the values specified.",
+				"operationId": "QuotaRequest_Create",
+				"x-ms-examples": {
+					"Quotas_Request_listForCompute": {
+						"$ref": "./examples/putComputeOneSkuQuotaRequest.json"
+					}
+				},
+				"x-ms-long-running-operation": true,
+				"x-ms-long-running-operation-options": {
+					"final-state-via": "location"
+				},
+				"parameters": [
+					{
+						"$ref": "#/parameters/SubscriptionIdInParameters"
+					},
+					{
+						"$ref": "#/parameters/ProviderIdInParameters"
+					},
+					{
+						"$ref": "#/parameters/LocationIdInParameters"
+					},
+					{
+						"$ref": "#/parameters/ResourceNameInParameters"
+					},
+					{
+						"$ref": "#/parameters/ApiVersionParameter"
+					},
+					{
+						"name": "createQuotaRequest",
+						"in": "body",
+						"description": "Quota requests payload.",
+						"required": true,
+						"schema": {
```

Comment:
request body and response body must use the same model. Any properties which is not settable by the end customer can be marked as readonly

Reply:
The Response body has the proviionState field inside the properties. We need to communicate the status of the quota request. The fields have bee marked as read-only.

Reply:
Based on - https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creating-or-updating-resources 
Changed to 201 with response in following format as documented  -
{
  "location": "North US",
  "tags": {
    "key1": "value 1",
    "key2": "value 2"
  },

  "properties": {
    "provisioningState": "Succeeded",
    "comment": "Resource defined structure"
  },
}

Reply:
@rahuls-microsoft  it is not addressed, as request and response is not matching. Request body, 200 & 201 response must have the same payload. why do we have three definitions ?

What is the async pattern used for 201 ? If the service is using provisioningState based polling, then please update description .(also same as request body).

Mark provisioningState as readonly property

Reply:
I will update the definitions to match and add  provisioningState as readonly property

Reply:
Updated it have same type for request and response.